### DISCUSSION TITLE
Change Request: resolveConfigsRelativeTo

### DISCUSSION THREAD

**Discussion by onlywei** [eslint/eslint#18415] [Q&A]
### ESLint version

8.57.0

### What problem do you want to solve?

ESLint currently has a `resolvePluginsRelativeTo` parameter in its constructor, but does not have a `resolveConfigsRelativeTo` param.

**Why do I want this?**
I have a monorepo with the current file structure:

```
apps/
-  app1/
  -  package.json
   .............
-  app127/
  -  package.json
packages/
-  package1/
  -  package.json
   ............
-  package453/
  -  package.json
repotools/
-  central-linter/
  -  package.json
.eslintrc.json
.eslintignore
```

I want to install `eslint` and all of its configs and plugins into the `repotools/central-linter/node_modules`. I do not want to install a separate copy of eslint and its plugins in each app/package. Also note that there is no `package.json` file at the root of the repo because I have lots of issues with npm workspaces.

I also want to keep the `.eslintrc.json` and `.eslintignore` files at the root of the monorepo for better IDE integration.

I am able to accomplish all of this **except** for resolving the shared `eslint-config-*` modules, since they are still resolving relative to the repository root.

### What do you think is the correct solution?

Add a `resolveConfigsRelativeTo` that works the same way as `resolvePluginsRelativeTo` for extending shared configs.

### Participation

- [X] I am willing to submit a pull request for this change.

**Comment by aladdin-add** [comment]
The option is eslintrc-only, however, it's deprecated and feature-frozen.

**Reply by onlywei** [reply]
Can you help point me to any discussions about that?

**Reply by aladdin-add** [reply]
https://eslint.org/blog/2023/10/flat-config-rollout-plans/
https://eslint.org/blog/2022/08/new-config-system-part-3/#main
https://eslint.org/docs/latest/use/command-line-interface#--resolve-plugins-relative-to

**Reply by onlywei** [reply]
@aladdin-add Hmmm, I still want this feature and I cannot upgrade to flat config or ESLint v9 yet because some plugins and parsers that I use don't support ESLint v9 yet. I made this PR for now: https://github.com/eslint/eslintrc/pull/159
